interfaces/builtin/maliit_input_method.go

```diff
@@ -0,0 +1,152 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var maliitInputMethodPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as maliit server.
+
+# DBus accesses
+#include <abstractions/dbus-session>
+
+# allow binding to the maliit dbus server for address negotiation
+dbus (bind)
+    bus=session
+    name="org.maliit.server",
+
+/usr/share/glib-2.0/schemas/ r,
+
+# maliit uses peer-to-peer dbus over a unix socket after address negotiation
+unix (bind, send, receive, listen, accept) addr="@/tmp/maliit-server/dbus-*",
+`)
+
+var maliitInputMethodConnectedSlotAppArmor = []byte(`
+# Provide access to the maliit address service
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var maliitInputMethodConnectedPlugAppArmor = []byte(`
+# Description: Allow applications to connect to a maliit socket
+# Usage: common
+
+#include <abstractions/dbus-session>
+
+# Find the maliit input method socket
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+unix (send, receive, connect) addr=none peer=(label=###SLOT_SECURITY_TAGS###),
```

Comment:
Is this 'type=stream'?

Also, I see 3 different IPC mechanisms:
1. org.maliit.server DBus bind in maliitInputMethodPermanentSlotAppArmor
2. org.maliit.Server.Address DBus interface in maliitInputMethodConnectedPlugAppArmor
3. @/tmp/maliit-server/dbus-* abstract socket bind/etc in maliitInputMethodPermanentSlotAppArmor
4. addr=none anonymous socket in maliitInputMethodConnectedPlugAppArmor

It seems that '1' and '2' go together, but the capitalization is different (ie, org.maliit.server in 'name' and 'path' in maliitInputMethodConnectedPlugAppArmor, but org.maliit.Server in 'interface'. If these go together, can you add a comment to maliitInputMethodConnectedSlotAppArmor and maliitInputMethodConnectedPlugAppArmor that says something about 'address negotiation' so it is understood that it goes back to the well-known DBus name you are binding to in maliitInputMethodConnectedSlotAppArmor?

'3' and '4' are different. There is this comment: "# maliit uses peer-to-peer dbus over a unix socket after address negotiation" but can you expand on that? Eg, "# maliit uses peer-to-peer dbus over a unix socket after address negotiation to do x, y and z". Furthermore, I expected to see a unix rule of this form in maliitInputMethodConnectedPlugAppArmor:
```
unix (send, receive, connect) type=stream peer=(addr=@/tmp/maliit-server/dbus-*, label=###SLOT_SECURITY_TAGS###),
```

Can you add a similar comment for the anonymous socket in '4'? Furthermore, I expected to see a unix rule of this form in maliitInputMethodConnectedSlotAppArmor:
```
unix (accept, receive, send) addr=none peer=(label=###PLUG_SECURITY_TAGS###),
```

Put another way, for each IPC mechanism, we should have a rule in ConnectedSlot to allow plugging clients to connect, a corresponding rule in ConnectedPlug to allow connecting to slotting server, and comments in both describing what goes where and why.